py-polars/polars/utils/various.py

```diff
@@ -375,29 +375,50 @@ def find_stacklevel() -> int:
 
 
 def _get_stack_locals(
```

Comment:
are we sure this can be used for #9958? doesn't it retrieve too many locals? wouldn't we only want the locals from the caller's frame?

Reply:
That's why I added the `n_frames` limiter so we don't go all the way up the stack. In conjunction with `named` and `of_type` we should be able to rapidly see if something with the given name and the expected type is in the most recent frame without going "to infinity... and beyond!" ;)

Reply:
> from the calling location.

ah, nice!

Reply:
sorry but this might still find things within polars? like, if there's
```python
my_dict = {1: 2}
pl.col('a').apply(lambda x: my_dict[x])
```
and within the polars source code there was also a `my_dict` local variable, which one would it pull?

Reply:
> within the polars source code there was also a my_dict local variable, which one would it pull?

I'll have to double-check in more detail and setup some funky tests to track frame/stack composition, but I suspect that even if there's a collision opportunity the chances are slim that somebody would both hit on a variable name that was in our stack AND it was a dict in their lambda - but let's see if we can minimise/guarantee it :)